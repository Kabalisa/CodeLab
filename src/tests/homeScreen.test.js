import HomeScreen from '../screens/homeScreen';

describe('HomeScreen tests', () => {
  const props = { navigation: { navigate: jest.fn() } };
  it('should render without error', () => {
    const wrapper = shallow(<HomeScreen {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should press navigate to login', () => {
    const wrapper = shallow(<HomeScreen {...props} />);
    const button = wrapper.find('TouchableOpacity');
    button.props().onPress();
  });
});
