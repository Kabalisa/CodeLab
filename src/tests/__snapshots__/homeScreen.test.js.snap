// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen tests should render without error 1`] = `
<View
  style={
    Object {
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#5075D4",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": "11%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "bottom": "2%",
          "color": "#5075D4",
          "fontSize": 24,
          "fontWeight": "bold",
          "left": "10%",
        }
      }
    >
      CodeLab
    </Text>
  </View>
  <View
    style={
      Object {
        "height": "89%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginHorizontal": "9%",
          "top": "10%",
          "zIndex": 2,
        }
      }
    >
      Find Java Developers In Lagos
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "marginHorizontal": "43%",
          "top": "73%",
          "zIndex": 2,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "width": 35,
            "zIndex": 2,
          }
        }
      />
    </TouchableOpacity>
    <View
      style={
        Object {
          "height": "100%",
          "position": "absolute",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": "100%",
            "opacity": 0.7,
          }
        }
      />
    </View>
  </View>
</View>
`;
